### PR TITLE
Unsolicited notifications

### PR DESCRIPTION
**Pull request by gvkalra**
A notification = key + value

Fixes: https://github.com/joncol/jcon-cpp/issues/32

### CODE REVIEW

File: src/jcon/json_rpc_tcp_server.cpp [hunk 1]
```diff
@@ -54,6 +54,17 @@ JsonRpcEndpoint* JsonRpcTcpServer::findClient(QObject* socket)
...
+{
+  int size = static_cast<int>(m_client_endpoints.size());
+  QVector<JsonRpcEndpoint*> rpc_endpoints(size, nullptr);
+
+  for (auto const& client_endpoint : m_client_endpoints)
+    rpc_endpoints.append(client_endpoint.second.get());
+
+  return rpc_endpoints;
+}
+
```
Comment (joncol) [inline src/jcon/json_rpc_tcp_server.cpp:67]: Nitpick: Could you use 4 space-indentation for consistency?

File: src/main.cpp [hunk 2]
```diff
@@ -218,4 +239,17 @@ int main(int argc, char* argv[])
...
+      auto rpc_client = startClient(&app, SocketType::tcp, true);
+      QObject::connect(rpc_client, &jcon::JsonRpcClient::notificationReceived,
+                       &app, [](const QString& key, const QVariant& value){
+                         qDebug() << "Received notification:"
+                                  << "Key:" << key
+                                  << "Value:" << value;
+                       });
+      app.exec();
+      delete server;
+    }
```
Comment (joncol) [inline src/main.cpp:254]: 4 space indentation please :)

File: src/notification_service.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    QMap<QString, QVariant> map;
+    map["one"] = 1;
+    map["three"] = 3;
+    map["seven"] = 7;
+    emit sendUnsolicitedNotification("myMap", map);
+
+    emit sendUnsolicitedNotification("myDouble", 54.53);
+  });
+  m_timer->start();
+}
```
Comment (joncol) [inline src/notification_service.cpp:24]: 4 space indentation please :)

File: src/notification_service.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public:
+  NotificationService(QObject *parent = nullptr);
+  virtual ~NotificationService();
+
+signals:
+  void sendUnsolicitedNotification(const QString&, const QVariant&);
+
+private:
+  QTimer *m_timer;
+};
```
Comment (joncol) [inline src/notification_service.h:19]: 4 space indentation please :)

File: src/jcon/json_rpc_tcp_server.cpp (hunk 1 above)
Comment (gvkalra) [reply src/jcon/json_rpc_tcp_server.cpp:67]: sorry about this. I'll do it :+1:

**Comment by joncol** [comment]
Thank you for the PR!

**Pull request merged**